src/optimizer/query_to_operator_transformer.cpp

```diff
@@ -245,11 +243,72 @@ void QueryToOperatorTransformer::Visit(parser::InsertStatement *op) {
     op->select->Accept(this);
     insert_expr->PushChild(output_expr_);
     output_expr_ = insert_expr;
+    return;
+  }
+  // column_objects represents the columns for the current table as defined in
+  // its schema
+  auto column_objects = target_table->GetColumnObjects();
+  // INSERT INTO table_name VALUES (val1, val2, ...), (val_a, val_b, ...), ...
+  if (op->columns.empty()) {
+    for (const auto &values : op->insert_values) {
+      if (values.size() > column_objects.size()) {
+        throw CatalogException(
+            "ERROR:  INSERT has more expressions than target columns");
+      } else if (values.size() < column_objects.size()) {
+        for (oid_t i = values.size(); i != column_objects.size(); ++i) {
+          // check whether null values are allowed in the rest of the columns
```

Comment:
@pmenon I think I can't replace this loop with a range-based equivalent because `column_objects` is a `HashMap` instead of a vector, and I need to start the index `i` in the middle.

Otherwise, I've replaced all the loops you mentioned with range-based ones.